src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable} from '@angular/core';
+import {OverlayContainer} from '@angular/cdk/overlay';
+
+const isDarkThemeKey = 'isDarkTheme';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
```

Comment:
I'm not sure we need a whole service just to toggle a class on the body.

Reply:
I agree, but I feel like using a service to access `localStorage` is the proper solution.

Reply:
It's just two extra function calls though (one for saving and one for reading). It's not like we need to mock this out in tests or have access to it anywhere else in the app.

Reply:
I kind of agree with Kristiyan here. Mostly because it's another file that needs to be loaded for every demo page load (which can be just avoided)